mmcv/runner/hooks/optimizer.py

```diff
@@ -34,6 +34,27 @@ def after_train_iter(self, runner):
         runner.optimizer.step()
 
 
+@HOOKS.register_module()
+class GradientCumulativeOptimizerHook(OptimizerHook):
+
+    def __init__(self, grad_clip=None, cumulative_iters=1):
+        super(GradientCumulativeOptimizerHook, self).__init__(grad_clip)
+        self.cumulative_iters = cumulative_iters
+
+    def after_train_iter(self, runner):
+        runner.outputs['loss'] = runner.outputs['loss'] / self.cumulative_iters
```

Comment:
I'm confusing about the detailed implementation of this function. Could you please offer a reference for this implementation to show that it is a **general case**? 

In my opinion, the accumulative gradients are adopted to avoid large batch sizes. Is it right? If it's right, why should we divide the `self.cumulative_iters`?

Reply:
Gradient accumulate is to achieve an equivalent larger batch size with small batch size. Therefore, the loss should be normalised. See more at https://discuss.pytorch.org/t/pytorch-gradient-accumulation/55955/2

Reply:
Ok, got it. You may specify this in the docs.

In addition, please also fix the corner case where `total_iters % cumulative_iters != 0`

Reply:
Hello, isn't BatchNormalization an issue as mentioned in here https://gist.github.com/thomwolf/ac7a7da6b1888c2eeac8ac8b9b05d3d3#gistcomment-3381285

Reply:
@ggalan87 Yes the behavior of batch normal is different, however, not all network contain batchnorm